src/Creator/InvoiceForOrderCreator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\InvoicingPlugin\Creator;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\InvoicingPlugin\Entity\InvoiceInterface;
+use Sylius\InvoicingPlugin\Generator\InvoiceGeneratorInterface;
+use Sylius\InvoicingPlugin\Repository\InvoiceRepository;
+
+final class InvoiceForOrderCreator implements InvoiceForOrderCreatorInterface
+{
+    /** @var InvoiceRepository */
+    private $invoiceRepository;
+
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    /** @var InvoiceGeneratorInterface */
+    private $invoiceGenerator;
+
+    public function __construct(
+        InvoiceRepository $invoiceRepository,
+        OrderRepositoryInterface $orderRepository,
+        InvoiceGeneratorInterface $invoiceGenerator
+    ) {
+        $this->invoiceRepository = $invoiceRepository;
+        $this->orderRepository = $orderRepository;
+        $this->invoiceGenerator = $invoiceGenerator;
+    }
+
+    public function __invoke(string $orderNumber, \DateTimeInterface $dateTime): void
+    {
+        /** @var InvoiceInterface|null $invoice */
+        $invoice = $this->invoiceRepository->findOneByOrderNumber($orderNumber);
+
+        if (null !== $invoice) {
+            return;
```

Comment:
A quick idea - maybe we could throw an exception (sth like `InvoiceAlreadyGenerated`) and then react on it somehow a level up? Silently _not doing anything_ just seems buggy 😄 🚀